Riot/Modules/Room/CreationModal/RoomCreationEventsModal/RoomCreationEventsModalViewModel.swift

```diff
@@ -0,0 +1,157 @@
+// File created from ScreenTemplate
+// $ createScreen.sh Modal2/RoomCreation RoomCreationEventsModal
+/*
+ Copyright 2020 New Vector Ltd
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+
+final class RoomCreationEventsModalViewModel: RoomCreationEventsModalViewModelType {
+    
+    // MARK: - Properties
+    
+    // MARK: Private
+
+    private let session: MXSession
+    private let bubbleData: MXKRoomBubbleCellDataStoring
+    private let roomState: MXRoomState
+    private var currentOperation: MXHTTPOperation?
+    private lazy var eventFormatter: EventFormatter = {
+        return EventFormatter(matrixSession: self.session)
+    }()
+    private var events: [MXEvent] = []
+    
+    // MARK: Public
+
+    weak var viewDelegate: RoomCreationEventsModalViewModelViewDelegate?
+    weak var coordinatorDelegate: RoomCreationEventsModalViewModelCoordinatorDelegate?
+    
+    var numberOfRows: Int {
+        return events.count
+    }
+    func rowViewModel(at indexPath: IndexPath) -> RoomCreationEventRowViewModel? {
+        let event = events[indexPath.row]
+        let formatterError = UnsafeMutablePointer<MXKEventFormatterError>.allocate(capacity: 1)
+        if let string = eventFormatter.attributedString(from: event, with: roomState, error: formatterError) {
+            if string.string.hasPrefix("·") {
+                return RoomCreationEventRowViewModel(title: string)
+            }
+            let mutableString = NSMutableAttributedString(attributedString: string)
+            mutableString.insert(NSAttributedString(string: "· "), at: 0)
+            return RoomCreationEventRowViewModel(title: mutableString)
+        }
+        return RoomCreationEventRowViewModel(title: nil)
+    }
+    var roomName: String? {
+        if let name = roomState.name, name.count > 0 {
```

Comment:
Can we return `summary.displayname` directly?
This should be the display name used everywhere in the app.